std/shared/src/main/scala-2/AsyncAwait.scala

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2020-2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.effect.std
+
+import scala.annotation.compileTimeOnly
+import scala.reflect.macros.blackbox
+
+import cats.effect.kernel.Async
+import cats.effect.kernel.syntax.all._
+import cats.syntax.all._
+import cats.effect.kernel.Outcome.Canceled
+import cats.effect.kernel.Outcome.Errored
+import cats.effect.kernel.Outcome.Succeeded
+
+/**
+ * WARNING: This construct currently only works on scala 2 (2.12.12+ / 2.13.3+),
+ * relies on an experimental compiler feature enabled by the -Xasync
+ * scalac option, and should absolutely be considered unstable with
+ * regards to backward compatibility guarantees (be that source or binary).
+ *
+ * Partially applied construct allowing for async/await semantics,
+ * popularised in other programming languages.
+ *
+ * {{{
+ * object ioAsyncAwait extends AsyncAwaitDsl[IO]
+ * import ioAsyncAwait._
+ *
+ * val io : IO[Int] = ???
+ * async { await(io) + await(io) }
```

Comment:
Since everyone just copy/pastes from examples:

```suggestion
 * object dsl extends AsyncAwaitDsl[IO]
 * import dsl._
 *
 * val io: IO[Int] = ???
 * async { await(io) + await(io) }
```